Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -5917,6 +5917,9 @@ static bool hasVisibleBoxDecorationsOrBackground(const RenderElement& renderer)
 #if HAVE(SUPPORT_HDR_DISPLAY)
 static bool rendererHasHDRContent(const RenderElement& renderer)
 {
+    if (PermissionsPolicy::isFeatureEnabled(PermissionsPolicy::Feature::Tonemapping, renderer.document(), PermissionsPolicy::ShouldReportViolation::No))
```

Comment:
Does this work through nested frames? You need tests for testing, in both directions.